### DISCUSSION TITLE
Create primary-secondary interface pair from disconnected mesh

### DISCUSSION THREAD

**Discussion by chunhuizhao478** [idaholab/moose#25627] [Q&A General]
Hi all, I try to seek for other options **without** using "BreakMeshByBlockGenerator" to handle interface calculations:

I generate two mesh block (0 - top(grey), 1 - bottom(red) ) as follows

```
[Mesh]
    [generated0]
        type = GeneratedMeshGenerator
        dim = 2
        nx = 201
        ny = 200
        xmin = -10050
        xmax = 10050
        ymin = 0
        ymax = 10000
        boundary_name_prefix = pillar1
    []
    [rename_block0]
        type = RenameBoundaryGenerator
        input = generated0
        old_boundary = '0 1 2 3'
        new_boundary = 'block0_bottom block0_right block0_left block0_top'
    []
    [block0_id]
        type = SubdomainIDGenerator
        input = rename_block0
        subdomain_id = 0
    []
    [generated1]
        type = GeneratedMeshGenerator
        dim = 2
        nx = 201
        ny = 200
        xmin = -10050
        xmax = 10050
        ymin = -10000
        ymax = 0
        boundary_name_prefix = pillar2
        boundary_id_offset = 4
    []
    [rename_block1]
        type = RenameBoundaryGenerator
        input = generated1
        old_boundary = '4 5 6 7'
        new_boundary = 'block1_bottom block1_right block1_left block1_top'
    []
    [block1_id]
        type = SubdomainIDGenerator
        input = rename_block1
        subdomain_id = 1
    []
    [collect]
        type = MeshCollectionGenerator
        inputs = 'block0_id block1_id'
    []
[]
```
<img width="660" alt="Screenshot 2023-09-29 at 11 27 03 AM" src="https://github.com/idaholab/moose/assets/93891863/4383c867-1c4b-4bb0-ad3e-04572c99a481">

Now I would like to do calculations based on the two boundaries in the center, I know I could extract boundary information from both sides by using a InterfaceMaterial or InterfaceKernel, but in order to achieve it, I need to somehow assign primary-secondary information to link the two interfaces (like creating a "Block0_Block1" in what BreakMeshByBlockGenerator could achieve by splitting the single mesh, here I already have two interfaces), I'm not sure if it is feasible to do or not.

**Comment by lindsayad** [comment]
[SideSetsBetweenSubdomainsGenerator](https://mooseframework.inl.gov/source/meshgenerators/SideSetsBetweenSubdomainsGenerator.html) is what I used in the first interface kernel examples I created
